index.html

```diff
@@ -1,390 +1,1451 @@
+
+    <ol>
+      <li>Let |p:Promise| be [=a new promise=].
+      <li>
+        If [=this=].{{[[state]]}} is not `"opened"`, reject |promise| with an
+        "{{InvalidStateError}}" {{DOMException}} and return |promise|.
+      <li>Perform the following steps [=in parallel=]:
+        <ol>
+          <li>
+            Query the operating system for the status of the control signals
+            that may be asserted by the device connected to the serial port.
+          <li>
+            If the operating system fails to determine the status of these
+            signals for any reason, [=queue a task=] to reject |promise| with a
+            "{{NetworkError}}" {{DOMException}} on the [=serial port task
+            source=] and abort these steps.
+          <li>
+            Let |signals:SerialInputSignals| be a [=new=]
+            {{SerialInputSignals}}.
+          <li>
+            Set |signals|.{{SerialInputSignals/dataCarrierDetect}} to `true` if
+            the "data carrier detect" or "DCD" signal has been asserted by the
+            device, and `false` otherwise.
+          <li>
+            Set |signals|.{{SerialInputSignals/clearToSend}} to `true` if the
+            "clear to send" or "CTS" signal has been asserted by the device, and
+            `false` otherwise.
+          <li>
+            Set |signals|.{{SerialInputSignals/dataCarrierDetect}} to `true` if
+            the "ring indicator" or "RI" signal has been asserted by the device,
+            and `false` otherwise.
+          <li>
+            Set |signals|.{{SerialInputSignals/dataCarrierDetect}} to `true` if
+            the "data set ready" or "DSR" signal has been asserted by the
+            device, and `false` otherwise.
+          <li>
+            [=Queue a task=] to resolve |promise| with |signals| on the [=serial port
+            task source=].
+        </ol>
+      <li>Return |promise|.
+    </ol>
+
+    <section data-dfn-for="SerialInputSignals">
+    <h4>{{SerialInputSignals}} dictionary</h4>
+
+    <pre class="idl">
+      dictionary SerialInputSignals {required boolean dataCarrierDetect;
+      required boolean clearToSend; required boolean ringIndicator; required
+      boolean dataSetReady;};
+    </pre>
+
+    <dl>
+      <dt><dfn>dataCarrierDetect</dfn> member</dt>
+      <dd>Data Carrier Detect (DCD)</dd>
+      <dt><dfn>clearToSend</dfn> member</dt>
+      <dd>Clear To Send (CTS)</dd>
+      <dt><dfn>ringIndicator</dfn> member</dt>
+      <dd>Ring Indicator (RI)</dd>
+      <dt><dfn>dataSetReady</dfn> member</dt>
+      <dd>Data Set Ready (DSR)</dd>
+    </dl>
+    </section>
+  </section>
+
+  <section>
+    <h3><dfn>close()</dfn> method</h3>
+
+    <aside class="example">
+      When communication with the port is no longer required it can be closed
+      and the associated resources released by the system.
+
+      <p>
+      Calling `port.`{{SerialPort/close()}} implicitly invokes
+      {{SerialPort/readable}}`.`{{ReadableStream/cancel()}} and
```

Comment:
Nit: this would be clearer to me as `port.readable.cancel()` and `port.writable.abort()`.